core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -347,35 +346,26 @@ public Table loadTable(SessionContext context, TableIdentifier identifier) {
             tableFileIO(context, response.config()),
             tableMetadata);
 
-    TableIdentifier tableIdentifier = loadedIdent;
     BaseTable table =
         new BaseTable(
             ops,
             fullTableName(loadedIdent),
-            report -> reportMetrics(tableIdentifier, report, session::headers));
+            metricsReporter(paths.metrics(loadedIdent), session::headers));
     if (metadataType != null) {
       return MetadataTableUtils.createMetadataTableInstance(table, metadataType);
     }
 
     return table;
   }
 
-  private void reportMetrics(
-      TableIdentifier tableIdentifier,
-      MetricsReport report,
-      Supplier<Map<String, String>> headers) {
-    try {
-      reporter.report(report);
-      if (reportingViaRestEnabled) {
-        client.post(
-            paths.metrics(tableIdentifier),
-            ReportMetricsRequest.of(report),
-            null,
-            headers,
-            ErrorHandlers.defaultErrorHandler());
-      }
-    } catch (Exception e) {
-      LOG.warn("Failed to report metrics to REST endpoint for table {}", tableIdentifier, e);
+  private MetricsReporter metricsReporter(
```

Comment:
rather than wrapping the combined reporter in a lambda, we just return the combined reporter here.